middleware/logging.go

```diff
@@ -95,15 +113,11 @@ var Logging = Log{
 	Log: logging.Global(),
 }
 
-func dumpRequest(req *http.Request) ([]byte, error) {
+func dumpRequest(req *http.Request, httpHeadersToExclude map[string]bool) ([]byte, error) {
 	var b bytes.Buffer
 
 	// Exclude some headers for security, or just that we don't need them when debugging
-	err := req.Header.WriteSubset(&b, map[string]bool{
-		"Cookie":        true,
-		"X-Csrf-Token":  true,
-		"Authorization": true,
-	})
```

Comment:
The `middleware.Log` struct is exported so users might be initialising it directly rather than just using the default instance created by the server (e.g. if they wanted to modify a config that's not exposed as a flag). In that case, if we merge this change, these users would have `HttpHeadersToExclude` set to an empty slice and the default headers will be logged. I think this library is widely used enough that we need to be careful when making changes like that. Therefore we should always remove the default headers, even if they're not in the `HttpHeadersToExclude` slice. 

@bboreham Do you think this would be an issue in reality?

Reply:
In case this is an issue I've created this commit https://github.com/weaveworks/common/pull/277/commits/65d65649d8d366de8cfd801db246d0d081a230fb

Reply:
@fionaliao yes, this is a good point. Thanks!